src/signer_node/message_processor/process_candidateblock.rs

```diff
@@ -56,6 +66,29 @@ where
     }
 }
 
+fn verify_aggregated_public_key<T>(
```

Comment:
I think this function should return an error when the block height of candidate block is scheduled to have aggregate public key but the candidate block doesn't have aggregate public key.

Otherwise, below wrong scenario occur.

Assumption:
* The candidate block height is 99.
* The node who receiving candidate block has federation config which block height is 100.( it means block 99 should have aggregate public key)
* The node is member of block_height 99 round.

Scenario: 

1. The node receives height 99 candidate block which doesn't have aggregated public key.
2. The node goes ahead signing process.(with don't make it error)
2. Success to proof generation.
2. Next round gets started. And the node receives height 100 candidate block. 
3. The node goes ahead signing process with assuming the current federation is new federation which would be started from 100 height.
4. Proof generation for the100 height candidate block is failure.

On this scenario, a single malicious node can prevent proof generation until other nodes remove federation config for the canceled federation.

Reply:
This function should return Error::InvalidAggregatedPublicKey if 

1. the block received from the other signer(master) does not have an aggregated public key
and 
2. this signer plans to change the federation at the next block.

Reply:
It was already implemented. sorry.